File: Extras/exercício_2-1-3.dart
import 'dart:math';

List lista = [];

List generateList(List l, int tamanho) {
  var rnd = Random();
  for (var i = 0; i <= tamanho; i++) {
    var randNumber = rnd.nextInt(30);
    if (l.length == 0) {
      l.add(randNumber);
    }
    if (!l.contains(randNumber)) {
      l.add(randNumber);
    }
    if (i == tamanho && l.length < tamanho) {
      i--;
    }
  }
  print(l);
  return l;
}

buscaLinear(List l, int j) {
  int i = 0;
  while (i < l.length && l[i] != j) {
    i++;
  }
  if (i != l.length) {
    return 'O resultado da pesquisa ${l[i]}, foi encontrado no índice ${i}';
  }
  return 'Valor $j não encontrado';
}

void main() {
  print(buscaLinear(generateList(lista, 20), 5));
}

File: Extras/exercício_2-1-3.js
var lista = []


function generateList(t) {

    for (var i = 0; i < t; i++) {
        var randonNumber = Math.floor(Math.random() * 20) + 1
        if (lista.length == 0) {
            lista.push(randonNumber)
        }
        if (!lista.includes(randonNumber)) {
            lista.push(randonNumber)
        }
        if (i == 4 && lista.length < t) {
            i--
        }
    }
    return lista;
}

function buscaLinear(l, num) {
    var i = 0

    while (i < l.length && l[i] != num) {
        i++
    }
    if (i != l.length) {
        console.log(l)
        console.log(`Resultado da pesquisa ${l[i]}, encontrado no índice ${i}`)
        return true
    }
    console.log('Número não encontrado')
    return false
}

console.log(buscaLinear(generateList(10), 15))